src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
+using Microsoft.CodeAnalysis;
+using WellKnownType = Microsoft.AspNetCore.App.Analyzers.Infrastructure.WellKnownTypeData.WellKnownType;
+
+namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel;
+
+internal class EndpointParameter
+{
+    public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    {
+        Type = parameter.Type;
+        Name = parameter.Name;
+        Source = EndpointParameterSource.Unknown;
+
+        if (GetSpecialTypeCallingCode(Type, wellKnownTypes) is string callingCode)
```

Comment:
OK, so thinking about emitting the TryParse/BindAsync work that I'm looking at, this is probably the point at which I would detect whether the parameter symbol implements IParsable or implements TryParse otherwise and evaluate whether by convention or attribute this value should attempted to be parsed .... and then just spit out the TryParse code.

Reply:
Basically. We might need to store statements we plan to run prior to invoking the handler that set local variables used in the `handler({argumentList})` invocation or to determine we should respond with a 400 response without invoking the handler. Take a look at `RequestDelegateFactory.CreateParamCheckingResponseWritingMethodCall` and how `ExtraLocals` and `ParamCheckExpressions` are built up if you haven't already. It's roughly analogous.

https://github.com/dotnet/aspnetcore/blob/fbdf6ab497ced552a01e38d570e9b2970030eb8b/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L859-L947